### PR TITLE
PROD-908 #1563-fix-WordPress database error Unknown column meta_id when BuddyBoss is active on Multisite

### LINKED ISSUE #1563: "WordPress database error Unknown column 'meta_id'" when BuddyBoss is active on Multisite
The issue is described here: https://buddypress.trac.wordpress.org/ticket/7834

Steps to reproduce:
1. Activate the BuddyBoss plugin on Multisite
2. Execute
```php
update_site_meta( get_current_blog_id(), 'delete_1', rand() );          
```
                                                                       
The "debug.log" file shows the following:
```
WordPress database error Unknown column 'meta_id' in 'field list' for query SELECT meta_id FROM wp_bp_user_blogs_blogmeta WHERE meta_key = 'delete_1' AND blog_id = 1 made by update_site_meta, update_metadata
```
**Jira issue** : [PROD-908]

[PROD-908]: https://buddyboss.atlassian.net/browse/PROD-908?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by meetmakadiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1563  .

### How to test the changes in this Pull Request:

1. compatible code doing with the BuddyPress  https://buddypress.trac.wordpress.org/attachment/ticket/7834/7834.bp_blog.patch

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — APPROVED** [review]

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@meetmakadiya there is no table `bp_blogmeta` on the platform. How this query will run `$wpdb->get_col( $wpdb->prepare( "SELECT meta_key FROM {$wpdb->bp_blogmeta} WHERE blog_id = %d", $blog_id ) );`???

CC @dipskakadiya

**Comment by dipskakadiya** [comment]
> @meetmakadiya there is no table `bp_blogmeta` on the platform. How this query will run `$wpdb->get_col( $wpdb->prepare( "SELECT meta_key FROM {$wpdb->bp_blogmeta} WHERE blog_id = %d", $blog_id ) );`???
> 
> CC @dipskakadiya

@chetansatasiya 
Actually `bp_blogmeta` refering to `bp_user_blogs_blogmeta ` table check below line. 

https://github.com/buddyboss/buddyboss-platform/blob/07dda455e1afcc8a4974a3c061307c2b28901407/src/bp-blogs/classes/class-bp-blogs-component.php#L65

File: src/bp-blogs/bp-blogs-cache.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ function bp_blogs_update_meta_cache( $blog_ids = false ) {
 		'object_ids'    => $blog_ids,
 		'object_type'   => buddypress()->blogs->id,
 		'object_column' => 'blog_id',
-		'cache_group'   => 'blog_meta',
+		'cache_group'   => 'bp_blog_meta',
```
Comment (dipskakadiya) [inline src/bp-blogs/bp-blogs-cache.php:30]: Please update `bp_blog` to `bb_blog`. We should have`bb` prefix.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-blogs/bp-blogs-cache.php (hunk 1 above)
Comment (meetmakadiya) [reply src/bp-blogs/bp-blogs-cache.php]: @dipskakadiya updated the PR as per request.

**Review by dipskakadiya — APPROVED** [review]

**Review by KartikSuthar — APPROVED** [review]

File: src/bp-blogs/bp-blogs-filters.php [hunk 2]
```diff
@@ -297,3 +297,30 @@ function bb_blogs_comment_mention_notification( $activity_id, $comment, $activit
...
+ * @see   bp_blogs_update_blogmeta()
+ * @see   bp_blogs_add_blogmeta()
+ * @see   bp_blogs_delete_blogmeta()
+ * @see   bp_blogs_get_blogmeta()
+ *
+ * @param string $retval column name.
+ *
+ * @return string
+ */
+function bp_blogs_filter_meta_column_name( $retval ) {
```
Comment (jitendrabanjara1991) [inline src/bp-blogs/bp-blogs-filters.php:321]:
```suggestion
function bb_blogs_filter_meta_column_name( $retval ) {
```

File: src/bp-blogs/bp-blogs-functions.php [hunk 3]
```diff
@@ -1359,20 +1359,27 @@ function bp_blogs_delete_blogmeta( $blog_id, $meta_key = false, $meta_value = fa
...
+	add_filter( 'sanitize_key', 'bp_blogs_filter_meta_column_name' );
 
 	$retval = false;
 	foreach ( $keys as $key ) {
-		$retval = delete_metadata( 'blog', $blog_id, $key, $meta_value, $delete_all );
+		$retval = delete_metadata( 'bp_blog', $blog_id, $key, $meta_value, $delete_all );
 	}
 
 	remove_filter( 'query', 'bp_filter_metaid_column_name' );
+	remove_filter( 'sanitize_key', 'bp_blogs_filter_meta_column_name' );
```
Comment (jitendrabanjara1991) [inline src/bp-blogs/bp-blogs-functions.php:1382]:
```suggestion
	remove_filter( 'sanitize_key', 'bb_blogs_filter_meta_column_name' );
```

**Pull request merged**